### DISCUSSION TITLE
A2DP source stuttering

### DISCUSSION THREAD

**Discussion by thezenox** [pschatzmann/ESP32-A2DP#492] [Q&A] [answered]
I have issues with stuttering audio output in A2DP source mode. See audio attached.
[audio.zip](https://github.com/pschatzmann/ESP32-A2DP/files/12927929/audio.zip)
Using examples/bt_music_sender/bt_music_sender.ino
I just changed name of BT device to connect with.

esp32-arduino current Github master ([4b3aebe](https://github.com/espressif/arduino-esp32/commit/4b3aebe11fa5acc0f8b259105b4d1990b95c399b)) and current version of ESP32-A2DP ([10db711](https://github.com/pschatzmann/ESP32-A2DP/commit/10db71139b38605a4fb2b296b2c257f7797813b2))

Tested on ESP32-WROOM-32E and ESP32-WROVER-B, no difference. Tried different A2DP sinks (UE Boom, Echo Dot)

Any idea what could be wrong?

**Comment by pschatzmann** [comment, marked as answer]
Not sure, did you try to play with the delay in the callback of sketch: e.g. with delay(0), w/o it or just with a yied() ?

**Reply by thezenox** [reply]
With yield() and delay(0) it works fine, but watchdog triggers after ~20 seconds. delay(1) in get_data_frames() causes stuttering. Delay in loop() dosn't matter.

**Comment by pschatzmann** [comment]
So, if delay(1) is too much, maybe try it with delayMicroseconds().
Or you can try to call delay(1) only with every nth callback call....

**Comment by thezenox** [comment]
delayMicroseconds() does not reset wdt timer (as far as i know). If 
I completly disable the WDT on both cores and use yield(), I get far less stuttering, just once every ~20 seconds. Still not perfect but much better.
